src/main/java/emailvalidator4j/lexer/EmailLexer.java

```diff
@@ -16,7 +16,7 @@
 
     public void lex(String input) {
         Pattern pattern = Pattern.compile(
-                "(([a-zA-Z_]|[^\\u0000-\\u007F])+[46]?)|([0-9]+)|(\r\n)|(::)|(\\s+?)|(.)|(\\p{Cc}+)", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE
+                "(([a-zA-Z0-9!#$%&'*+\\-/=?^_`{|}~]|[^\\u0000-\\u007F])+[46]?)|([0-9]+)|(\r\n)|(::)|(\\s+?)|(.)|(\\p{Cc}+)", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE
```

Comment:
Expand pattern to gather longest possible spans of atext into single GENERIC tokens. Follows atext as defined in [RFC 5322, Section 3.2.3](https://tools.ietf.org/html/rfc5322#section-3.2.3):
```plain
   atext           =   ALPHA / DIGIT /    ; Printable US-ASCII
                       "!" / "#" /        ;  characters not including
                       "$" / "%" /        ;  specials.  Used for atoms.
                       "&" / "'" /
                       "*" / "+" /
                       "-" / "/" /
                       "=" / "?" /
                       "^" / "_" /
                       "`" / "{" /
                       "|" / "}" /
                       "~"
```

and extended to UTF-8 by [RFC 6531, Section 3.3](https://tools.ietf.org/html/rfc6531#section-3.3):
```plain
atext   =/  UTF8-non-ascii
    ; extend the implicit definition of atext in
    ; RFC 5321, Section 4.1.2, which ultimately points to
    ; the actual definition in RFC 5322, Section 3.2.3
```